http-client/src/main/java/io/airlift/http/client/jetty/JettyResponseFuture.java

```diff
@@ -62,6 +64,8 @@ private enum JettyAsyncHttpState
         this.span = requireNonNull(span, "span is null");
         this.stats = requireNonNull(stats, "stats is null");
         this.recordRequestComplete = recordRequestComplete;
+
+        jettyRequest.onRequestCommit(this);
```

Comment:
Register this on the outside, so that we don't leak a partially constructed object